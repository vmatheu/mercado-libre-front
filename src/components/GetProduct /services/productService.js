import axios from 'axios';
import { PRODUCT_SERVICE_ENDPOINT_ID } from '../../constants';

const transformProductSearch = (respuesta) => ({
    id: respuesta.item.id,
    title: respuesta.item.title,
    currency: respuesta.item.price.currency,
    amount: respuesta.item.price.amount,
    decimals: respuesta.item.price.decimals,
    picture: respuesta.item.picture,
    condition: respuesta.item.condition,
    freeShipping: respuesta.item.freeShipping,
    description: respuesta.item.description
})



export const findProductById = async (search) => {
  const productServiceEnpoint = PRODUCT_SERVICE_ENDPOINT_ID;
  const data = await axios.get(`${productServiceEnpoint}/${search}`)
    .then(response => ({
      item: transformProductSearch(response.data),
      categories: response.data.categories,
    }))
    .catch(error => {
      console.error(error)
      return []
    });
  return data;
};

export default { findProductById };
