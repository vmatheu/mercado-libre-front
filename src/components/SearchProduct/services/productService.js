import axios from 'axios';
import { PRODUCT_SERVICE_ENDPOINT } from '../constants';

export const findProductBySearch = async (search) => {
  const productServiceEnpoint = PRODUCT_SERVICE_ENDPOINT['development'].url;
  const data = await axios.get(`${productServiceEnpoint}${search}`)
    .then(response => (response.data))
    .catch(error => ([]));
  return data;
};

export default { findProductBySearch };
