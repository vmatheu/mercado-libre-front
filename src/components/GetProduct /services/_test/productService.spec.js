import { findProductBySearch } from '../productService';
import axios from 'axios';

jest.mock('axios');


const endpointResponse = {
  data: [
    {
      "author": {
      "name": "Victor",
      "lastname": "Matheu"
      },
      "item": {
        "id": "id_test",
        "title": "Flor",
        "price": {
          "currency": "CLP",
          "amount": 3000,
          "decimals": 0
        },
        "picture": "http://la_imagen",
        "condition": "condition",
        "free_shipping": true
      }
    }
  ]
}

describe('ProductService ', () => {
  beforeEach(() => {
    axios.get.mockImplementation(() => Promise.resolve(endpointResponse))
  });

  it('should be return object when calling getProductBySearch', async () => {
    const result = await findProductBySearch('1');
    expect(result).toMatchSnapshot();
  });
});
