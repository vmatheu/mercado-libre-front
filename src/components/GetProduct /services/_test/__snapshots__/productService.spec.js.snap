// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductService  should be return object when calling getProductBySearch 1`] = `
Object {
  "amount": 3000,
  "condition": "condition",
  "currency": "CLP",
  "decimals": 0,
  "description": undefined,
  "freeShipping": undefined,
  "id": "id_test",
  "picture": "http://la_imagen",
  "title": "Flor",
}
`;
